src/popover/popover.directive.ts

```diff
@@ -54,10 +66,11 @@ export class PopoverDirective implements OnInit, OnDestroy {
   @Output() public onHidden: EventEmitter<any>;
 
   private _popover: ComponentLoader<PopoverContainerComponent>;
+  private _outsideClickListener: Function;
 
-  public constructor(_elementRef: ElementRef,
-                     _renderer: Renderer,
-                     _viewContainerRef: ViewContainerRef,
+  public constructor(protected _elementRef: ElementRef,
```

Comment:
Um, access modifiers provide injected services as class fields. Without them I would have to declare and assign 3 variables more... E.g.

    constructor(protected _service: Service)
is an equivalent of

    protected _service: Service;
    constructor(_service: Service) {
        this._service = _service;
    }